src/System.Management.Automation/engine/Modules/NewModuleManifestCommand.cs

```diff
@@ -484,6 +484,10 @@ private string QuoteName(object name)
         {
             if (name == null)
                 return "''";
+            else if (name is Uri)
```

Comment:
Avoid casting multiple times. consider assigning name as Uri and testing for null.